src/components/comments.jsx

```diff
@@ -1,26 +1,47 @@
 import { getComments } from "../api";
 import { useState, useEffect } from "react";
 import Container from "react-bootstrap/Container";
+import PostComment from "./post-comment.jsx";
 
 const Comments = ({ article_id }) => {
   const [articleComments, setArticleComments] = useState([]);
+  const [err, setErr] = useState([])
 
   useEffect(() => {
+    fetchComments()
+  }, [article_id]);
+
+  const fetchComments = () => {
     getComments(article_id).then((comments) => {
       setArticleComments(comments);
-    });
-  }, []);
+      setErr(null)
+    }).catch((error) => {
+      setErr('error fetching comments')
+    })
+
+  }
+
+  const handlePostedComments = (newComment) => {
+    const commentObj = {comment_id: `temp-id-${Date.now()}`, body: newComment}
+    setArticleComments((prevComments) => [...prevComments, commentObj])
+    setErr(null)
+    fetchComments()
```

Comment:
You're doing both optimistic rendering and also fetching comments in one go. It would be better to stick to one or the other. I say posting a comment shouldn't be optimistic as it's better that the user knows straight away if something goes wrong with the post